### PR TITLE
feat(docs): add gateway concept doc & update retrieval doc

### LINKED ISSUE #1217: NFT.Storage gateway docs
Quick explainer in Retrieve section about NFT.Storage gateway and how to use.

### PR DESCRIPTION
**Pull request by yusefnapora**
This will close #1217, but it needs some more meat on the bones. I'll try to wrap it up for review on Monday.

### CODE REVIEW

File: packages/website/pages/docs/concepts/gateways.md [hunk 1]
```diff
@@ -0,0 +1,96 @@
+# IPFS Gateways
+
+NFT.Storage uses the [InterPlanetary File System (IPFS)](https://ipfs.io) as a key part of its [storage and retrieval infrastructure][concepts-decentralized-storage].
+
+The IPFS network is a peer-to-peer network of computers that share resources to efficiently provide content to anyone that requests it. Computers that join the IPFS network use a protocol called Bitswap to request and supply blocks of data using a hash-based Content Identifier (CID) to uniquely identify each block.
```
Comment (vasco-santos) [inline packages/website/pages/docs/concepts/gateways.md:5]: Should we add link to bitswap docs? https://docs.ipfs.io/concepts/bitswap/
This knowledge is not needed for using the gateway, but if we talk about it maybe we should point it for interested people to look into what happens behind the scenes

File: packages/website/pages/docs/concepts/gateways.md [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+The NFT.Storage gateway is effectively a caching layer that sits in front of several other public IPFS gateways and "races" them against each other to provide the fastest response possible to end-user requests.
+
+This layered architecture has several benefits over using the "downstream" gateways directly. By using CloudFlare Workers to cache requests at the edge, the gateway can serve popular content quickly, while taking load off of the public IPFS gateways and peer-to-peer network.
+
+Requests for content that are not cached are served by making parallel requests to multiple public IPFS gateways and returning the first valid response. This gives you the performance of the fastest downstream gateway, which may vary from time to time due to network conditions and the state of the gateway's own local cache. 
+
+Sending multiple requests also reduces the dependence on any single gateway and protects your app from experiencing degraded service if a gateway goes down for maintainence or has connectivity issues.
+
+
```
Comment (vasco-santos) [inline packages/website/pages/docs/concepts/gateways.md:31]:
```suggestion
```

**Review by vasco-santos — COMMENTED** [review]
Great work @yusefnapora ❤️ 

Left some suggestions inline

**Comment by dchoi27** [comment]
@yusefnapora let's incorporate @vasco-santos 's suggestions (if you agree) and push + deploy by EOW! tyty

**Comment by yusefnapora** [comment]
Sorry bout the lag on this, folks. Thanks for the review @vasco-santos!

**Review by dchoi27 — APPROVED** [review]
LGTM!

**Pull request merged**